controllers/fetchdata_controller.go

```diff
@@ -98,7 +101,7 @@ func FetchCronJob(name, namespace string, client client.Client) (*batchv1.CronJo
 	return cronJob, err
 }
 
-func NewCronJob(m *curatorv1alpha1.FetchData, scheme *runtime.Scheme) *batchv1.CronJob {
+func NewCronJob(m *curatorv1alpha1.FetchData, scheme *runtime.Scheme, db *pgx.Conn) *batchv1.CronJob {
```

Comment:
Passing a pgx.Conn parameter feels a bit heavy here given this function's purpose is centered around generating a new CronJob resource. Can we work towards a model where we are mounting the Secret as a volume and the environment variables are [reading from that secret reference](https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/#define-container-environment-variables-using-secret-data)?